soc: nordic: Use proper devicetree entries for clock frequency

nordicjm opened the pull request:
dts: bindings: timer: Add default clock frequency for nordic grtc
    
    Adds a default for the nordic grtc



    soc: nordic: Use proper devicetree entries for clock frequency
    
    Sets the SYS_CLOCK_HW_CYCLES_PER_SEC Kconfig from devicetree
    entries. Also fixes invalid configuration on nrf54h20 whereby it
    attempts to take the clock frequency from a peripheral that does
    not exist

nordicjm committed:
dts: bindings: timer: Add default clock frequency for nordic grtc

Adds a default for the nordic grtc

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
soc: nordic: Use proper devicetree entries for clock frequency

Sets the SYS_CLOCK_HW_CYCLES_PER_SEC Kconfig from devicetree
entries. Also fixes invalid configuration on nrf54h20 whereby it
attempts to take the clock frequency from a peripheral that does
not exist

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
dts: arm: nordic: Add default rtc1 clock frequency to nrf53 network core

Adds the default 32.768KHz clock frequency to this peripheral, as it is
used as the system timing source

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

masz-nordic commented on soc/nordic/nrf54l/Kconfig.defconfig:
```diff
@@ -46,4 +46,8 @@ config BUILD_OUTPUT_ADJUST_LMA
 endif # RISCV
 
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default $(dt_node_int_prop_int,/soc/peripheral@40000000/grtc@e2000,clock-frequency) if NRF_GRTC_TIMER && TRUSTED_EXECUTION_NONSECURE
```
This makes an assumption about the GRTC peripheral address, which does not scale well.
Proper way to address this would be to add `dt_nodelabel_int_prop_int` functionality and use it here with `grtc` label.

nordicjm replied:
Completely overlooked that function, will update

masz-nordic replied:
I meant this dt function should be implemented and then used here - it is not present now.

nordicjm committed:
scripts: kconfig: Add dt_nodelabel_int_prop function

Adds a function which can be used to get the integer value of a
devicetree property in Kconfig from a nodelabel

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

masz-nordic reviewed (approved):
Looks great, thanks!

rruuaanng reviewed (approved):
Looks good.
